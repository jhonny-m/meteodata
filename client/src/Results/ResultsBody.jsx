import React from 'react';
import PropTypes from 'prop-types';
import resultsBodyStyles from './ResultsBody.module.css';
import ResultsTable from '../Results/Table/ResultsTable';
import ResultsGraph from '../Results/Graph/ResultsGraph';
import {formatterTemperatures} from './TemperatureUtils';

function ResultsBody({results}){
	return (
		<div className={resultsBodyStyles.container}>
			<ResultsGraph results={formatterTemperatures(results)}/>
			<div className={resultsBodyStyles.tableContainer}>
				<ResultsTable results={formatterTemperatures(results)}/>
			</div>
		</div>
	);

}
ResultsBody.propTypes={
	results:PropTypes.array.isRequired,
};

export default ResultsBody;