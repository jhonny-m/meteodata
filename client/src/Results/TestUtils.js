export  const mockedResults = [
	{
		'cityName':'Porto',
		'sunriseDate':1602657904,
		'sunsetDate':1602698118,
		'temperature':285.97
	},
	{
		'cityName':'Braga',
		'sunriseDate':1602657882,
		'sunsetDate':1602698049,
		'temperature':285.93
	},
	{
		'cityName':'Aveiro',
		'sunriseDate':1602657885,
		'sunsetDate':1602698157,
		'temperature':287.64
	}
];