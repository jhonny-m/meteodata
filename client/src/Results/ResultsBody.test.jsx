import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import ResultsBody from './ResultsBody';
import {mockedResults} from './TestUtils';


test('ResultsBody',()=>{
	const container = ResultsBody({results:mockedResults});
	expect(container).toMatchSnapshot();
});