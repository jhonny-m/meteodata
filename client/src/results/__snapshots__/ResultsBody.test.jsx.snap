// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsBody 1`] = `
<div
  className="container"
>
  <ResultsGraph
    results={
      Array [
        Object {
          "cityName": "Porto",
          "sunriseDate": 1602657904,
          "sunsetDate": 1602698118,
          "temperature": 13,
        },
        Object {
          "cityName": "Braga",
          "sunriseDate": 1602657882,
          "sunsetDate": 1602698049,
          "temperature": 13,
        },
        Object {
          "cityName": "Aveiro",
          "sunriseDate": 1602657885,
          "sunsetDate": 1602698157,
          "temperature": 14,
        },
      ]
    }
  />
  <div
    className="tableContainer"
  >
    <ResultsTable
      results={
        Array [
          Object {
            "cityName": "Porto",
            "sunriseDate": 1602657904,
            "sunsetDate": 1602698118,
            "temperature": 13,
          },
          Object {
            "cityName": "Braga",
            "sunriseDate": 1602657882,
            "sunsetDate": 1602698049,
            "temperature": 13,
          },
          Object {
            "cityName": "Aveiro",
            "sunriseDate": 1602657885,
            "sunsetDate": 1602698157,
            "temperature": 14,
          },
        ]
      }
    />
  </div>
</div>
`;
